lib/iris/tests/unit/experimental/geovista/test_extract_unstructured_region.py

```diff
@@ -0,0 +1,152 @@
+
+    def test_empty_indices(self):
+        mock_polydata_scalars = {
+            VTK_CELL_IDS: np.arange(0),
+            VTK_POINT_IDS: np.arange(0),
+        }
+        self.mocked_polydata.__getitem__.side_effect = mock_polydata_scalars.__getitem__
+        with pytest.raises(
+            IndexError, match="No part of `polydata` falls within `region`."
+        ):
+            extract_unstructured_region(
+                self.cube_mesh, self.mocked_polydata, self.mocked_region
+            )
+
+    def test_recreate_mesh(self):
+        extracted_region = extract_unstructured_region(
+            self.cube_mesh, self.mocked_polydata, self.mocked_region
+        )
+        if self.cube_mesh.location == "face":
+            assert extracted_region.mesh is not None
+        else:
+            assert extracted_region.mesh is None
+
+    def test_new_mesh_coords(self):
+        extracted_region = extract_unstructured_region(
+            self.cube_mesh, self.mocked_polydata, self.mocked_region
+        )
+        if self.cube_mesh.location == "face":
+            mesh_coords = extracted_region.coords(mesh_coords=True)
+            np.testing.assert_array_equal(
+                mesh_coords[0].bounds,
+                [[1200, 1201, 1202, 1203], [1204, 1205, 1206, 1207]],
+            )
+            np.testing.assert_array_equal(mesh_coords[0].points, [3200, 3201])
+            np.testing.assert_array_equal(
+                mesh_coords[1].bounds,
+                [[1100, 1101, 1102, 1103], [1104, 1105, 1106, 1107]],
+            )
+            np.testing.assert_array_equal(mesh_coords[1].points, [3100, 3101])
+
+    def test_no_mesh(self, cube_2d):
+        cube = cube_2d
+        with pytest.raises(ValueError, match="Cube must have a mesh"):
+            extract_unstructured_region(cube, self.mocked_polydata, self.mocked_region)
+
+    def test_edge_location(self, cube_mesh_edge):
+        with pytest.raises(
+            NotImplementedError,
+            match=f"Must be on face or node." f" Found: {cube_mesh_edge.location}.",
+        ):
+            extract_unstructured_region(
+                cube_mesh_edge, self.mocked_polydata, self.mocked_region
+            )
+
+    def test_cube_and_poly_shapes_mismatch(self):
+        self.mocked_polydata.GetNumberOfCells.return_value = 4
+        self.mocked_polydata.GetNumberOfPoints.return_value = 16
+        polydata_length = ()
+        if self.cube_mesh.location == "face":
+            polydata_length = self.mocked_polydata.GetNumberOfCells()
+        elif self.cube_mesh.location == "node":
+            polydata_length = self.mocked_polydata.GetNumberOfPoints()
+        with pytest.raises(
+            ValueError,
+            match=f"The mesh on the cube and the polydata"
+            f"must have the same shape."
+            f" Found Mesh: {self.cube_mesh.shape[self.cube_mesh.mesh_dim()]},"
+            f" Polydata: {polydata_length}.",
+        ):
```

Comment:
Nice dedication here! For future reference: the `match=` condition can just be a section of the error. We use that a lot when the error has dynamic content and we want to avoid the effort that you have put in here.

No change required though 🙂